internal/vm/setup.go

```diff
@@ -76,6 +76,22 @@ _G["process_auto_fields"] = function(plugin)
   return json.encode(p)
 end
 
+_G["transform"] = function(plugin)
+  if type(plugin) ~= "string" then
+    return nil, "invalidate type, expected string"
+  end
+  local tbl, err = json.decode(plugin)
+  if err then
+    return nil, err
+  end
+
+  p, err = Plugins:transform(tbl, nil, "insert")
+  if err then
+    return nil, json.encode(err)
```

Comment:
in Lua (contrary to Go), the idiomatic way to signal an error isn't by a non-nil second value, but by the falsyness of the first.
```suggestion
  p, err = Plugins:transform(tbl, nil, "insert")
  if not p then
    return nil, json.encode(err or "failure to transform plugin")
```
(yes, there are exceptions, like `json.decode`, that can validly return `false` or `nil`.  but they're considered bad taste, and usually add a first "ok" value, like `pcall` does)

Reply:
Thanks Javier! Fixed